examples/classifier/CL2_apply_correlation_based_kmeans_clustering.py

```diff
@@ -0,0 +1,46 @@
+# Standard library imports
+import sys
+new_path = sys.path[0][:-len("/examples/timeseries")]
+sys.path.append(new_path)
+
+# Third party imports
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+# Local application imports
+from scifin import timeseries as ts
+from scifin import statistics as st
+from scifin import montecarlo as mc
+from scifin import classifier as cl
+
+# Generate random walk processes
+mc1 = mc.generate_series(n=100, names_base="RW-", series_model=ts.random_walk,
+                         start_date="2000-01-01", end_date="2020-01-01", frequency='M',
+                         start_value=1., sigma=0.01)
+
+# Plot
+ts.multi_plot(mc1)
+
+# Build their covariance matrix
+cov1 = st.covariance_from_ts(list_ts=mc1, min_periods=10)
+corr1 = st.covariance_to_correlation(cov1)
+
+# Plot this correlation matrix
+sns.heatmap(corr1, cmap='viridis')
+plt.title("Initial correlation matrix")
+plt.show()
+
+# Apply KMeans base clustering
+clustered_corr1, clusters1, silh_score1 = cl.kmeans_base_clustering(corr1, max_num_clusters=10, n_init=10)
+
+# Plot the clustered correlation matrix
+sns.heatmap(clustered_corr1, cmap='viridis')
+plt.title("Clustered correlation matrix")
+plt.show()
+
+# Print clusters content
+print()
+labels1 = cl.convert_clusters_into_list_of_labels(mc1, clusters1)
+cl.print_clusters_content(mc1, labels1)
+plt.show()
+
```

Comment:
How are we supposed to use this in practice? When I run it in the terminal I don't see any plot because:
```
UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure
```

Reply:
I run this file from terminal and don't get any warning. I tried on Poetry shell and got no appearance of this warning (I get warnings from Pandas though).

The function convert_clusters_into_list_of_labels converts a dictionary of time series names gathered by cluster number into a list of labels, following the order of the list of time series (here mc1), and containing the cluster labeling numbers.

What can I do to help solving the problem on your side?

Reply:
I found a related question here https://stackoverflow.com/questions/56656777/userwarning-matplotlib-is-currently-using-agg-which-is-a-non-gui-backend-so.
It looks like I have to install some system wide dependency to make this work. I will try to figure this out at some point. Don't consider this issue as a blocking point for merging your PR

Reply:
Ok, so I installed the ``python3-tk`` package on my linux and it works now.
``sudo apt-get install python3-tk``

Reply:
Great. Glad it works now.